### PR TITLE
Update lettre to 0.10.0-alpha.1

### LINKED ISSUE #2629: mail: in local dev, JSON written to /tmp/ serializes message as decimal bytes
In local test mode, if you don't set up a mailgun account, emails get written as JSON to /tmp. This is useful and important for, e.g. verifying an email address to test local uploads.

However, the JSON that gets written is hard to use because the message body is encoded as an array of bytes in decimal:

{"envelope":{"forward_path":["example@hoffman-andrews.com"],"reverse_path":"test@localhost"},"message_id":"1c9aeff3-f174-495b-bcca-8062b6fb47c3","message":[83,117,98,106,101,99,116,58,32,80,108,101,97,115,101,32,99,111,110,102,105,114,109,32,121,111,117,114,32,101,109,97,105,108,32,97,100,100,114,101,115,115,13,10,84,111,58,32,60,106,115,104,97,50,64,104,111,102,102,109,97,110,45,97,110,100,114,101,119,115,46,99,111,109,62,13,10,70,114,111,109,58,32,60,116,101,115,116,64,108,111,99,97,108,104,111,115,116,62,13,10,68,97,116,101,58,32,83,117,110,44,32,49,50,32,74,117,108,32,50,48,50,48,32,49,52,58,48,57,58,51,51,32,45,48,55,48,48,13,10,77,73,77,69,45,86,101,114,115,105,111,110,58,32,49,46,48,13,10,77,101,115,115,97,103,101,45,73,68,58,32,60,49,99,57,97,101,102,102,51,45,102,49,55,52,45,52,57,53,98,45,98,99,99,97,45,56,48,54,50,98,54,102,98,52,55,99,51,46,108,101,116,116,114,101,64,108,111,99,97,108,104,111,115,116,62,13,10,13,10,72,101,108,108,111,32,106,115,104,97,50,33,32,87,101,108,99,111,109,101,32,116,111,32,67,114,97,116,101,115,46,105,111,46,32,80,108,101,97,115,101,32,99,108,105,99,107,32,116,104,101,10,108,105,110,107,32,98,101,108,111,119,32,116,111,32,118,101,114,105,102,121,32,121,111,117,114,32,101,109,97,105,108,32,97,100,100,114,101,115,115,46,32,84,104,97,110,107,32,121,111,117,33,10,10,104,116,116,112,115,58,47,47,99,114,97,116,101,115,46,105,111,47,99,111,110,102,105,114,109,47,82,70,113,109,104,77,57,49,110,86,76,55,81,78,83,76,112,101,85,71,119,104,72,114,103,76,13,10]}

It would be more useful to encode this as a string.

### PR DESCRIPTION
**Pull request by jsha**
Fixes #2629

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -62,7 +62,7 @@ tempfile = "3"
 parking_lot = "0.11"
 jemallocator = { version = "0.3", features = ['unprefixed_malloc_on_supported_platforms', 'profiling'] }
 
-lettre = "0.9"
+lettre = "0.10.0-alpha.1"
```
Comment (paolobarbolini) [inline Cargo.toml:65]: lettre uses rustls by default

```suggestion
lettre = { version = "0.10.0-alpha.1", default-features = false, features = ["file-transport", "smtp-transport", "native-tls", "hostname", "builder"] }
```

File: src/email.rs [hunk 2]
```diff
@@ -37,19 +34,17 @@ fn build_email(
...
         .as_ref()
         .map(|s| s.smtp_login.as_str())
         .unwrap_or("test@localhost");
 
-    let email = Email::builder()
-        .to(recipient)
-        .from(sender)
+    let email = Message::builder()
+        .to(recipient.parse().unwrap())
+        .from(sender.parse().unwrap())
```
Comment (paolobarbolini) [inline src/email.rs:45]: This should fix the `sending_to_invalid_email_fails` test
```suggestion
        .to(recipient.parse()?)
        .from(sender.parse()?)
```

File: Cargo.toml [hunk 3]
```diff
@@ -62,7 +62,7 @@ tempfile = "3"
 parking_lot = "0.11"
 jemallocator = { version = "0.3", features = ['unprefixed_malloc_on_supported_platforms', 'profiling'] }
 
-lettre = "0.9"
+lettre = { version = "0.10.0-alpha.1", default-features = false, features = ["file-transport", "smtp-transport", "native-tls", "hostname", "builder"] }
 lettre_email = "0.9"
```
Comment (JohnTitor) [inline Cargo.toml:66]: We can drop this now.

**Comment by JohnTitor** [comment]
I'm not sure whether it's actually unstable or not; r? @jtgeibel I'd defer the decision to Justin.

File: Cargo.toml (hunk 1 above)
Comment (jsha) [reply Cargo.toml]: Out of curiosity, does anything break if cargo uses rustls to talk to Mailgun?

Comment (paolobarbolini) [reply Cargo.toml]: rustls is fine, although it currently doesn't get tested as much as native-tls in lettre. I just pointed out you probably don't want multiple TLS backends.

File: src/email.rs [hunk 4]
```diff
@@ -113,21 +108,20 @@ fn send_email(recipient: &str, subject: &str, body: &str) -> AppResult<()> {
 
     match mailgun_config {
         Some(mailgun_config) => {
-            let mut transport = SmtpClient::new_simple(&mailgun_config.smtp_server)?
+            let transport = SmtpTransport::builder(&mailgun_config.smtp_server)
```
Comment (paolobarbolini) [inline src/email.rs:111]:
```suggestion
            let transport = SmtpTransport::relay(&mailgun_config.smtp_server)
```

I saw your issue again https://github.com/lettre/lettre/issues/431 and this needs to be a relay, else it won't be configured by default to speak TLS. Sorry for not catching this earlier. We'll update the lettre docs to better show this.

**Comment by jtgeibel** [comment]
To test on staging I had to rebase (in order to pick up #2657) and since there were merge conflicts in `Cargo.lock` I've pushed that branch here as well.

On staging I was able to update my email address and receive a new confirmation email. These changes look good to me, thanks @jsha and @paolobarbolini!

@bors r+

**Pull request merged**